index.js

```diff
@@ -206,6 +207,10 @@ function normalizeArguments(url, opts) {
 		}
 	}
 
+	if (isURL(url)) {
+		url = urlParseLax(url.href);
```

Comment:
There's no auth (`username`,`password`) check in this block like there is for string input.